### DISCUSSION TITLE
Flag order for trimmed line with max length

### DISCUSSION THREAD

**Discussion by misaki-web** [BurntSushi/ripgrep#2456] [General] [answered]
Is there a way to make ripgrep apply the `--trim` flag before reducing the line length with `--max-columns-preview -M`?

Here's an example of the current result (the string starts with 5 spaces):

```bash
$ echo "     0123456789abcdefghijklmnopqrstuvwxyz" | rg --trim --max-columns-preview -M 8 abc
012 [... 1 more match]
```

ripgrep reduces the line length, then it trims it. I expected the opposite, like this:

```bash
$ echo "     0123456789abcdefghijklmnopqrstuvwxyz" | rg --trim --max-columns-preview -M 8 abc
01234567 [... 1 more match]
```

**Comment by BurntSushi** [comment, marked as answer]
It does seem like `--trim` should just run before `-M`. I created an issue to investigate that: #2458. Ideally, we can just fix it to be more sensible. I do not want to get into the business of supporting arbitrary orderings of output format options.

I can't think of any work-around at present.
